R/sample.ballbin.R

```diff
@@ -1,416 +1,425 @@
+
+  #Limit plot size to 100 plots
+  if (obs > MAX.PLOTS) {
+    warning(paste0('There were more than ', MAX.PLOTS, ' samples --- plot shows only the first ', MAX.PLOTS, ' samples'))
+    ALLOC  <- ALLOC[1:MAX.PLOTS, ]
+    SAMPLE <- SAMPLE[1:MAX.PLOTS, ]
+    obs    <- MAX.PLOTS }
+
+  #Check installed packages and load them
+  GGPLOT2 <- requireNamespace('ggplot2', quietly = TRUE)
+  GREXTRA <- requireNamespace('gridExtra', quietly = TRUE)
+  if (!GGPLOT2) { stop('Error: Plotting a \'ballbins\' object requires the ggplot2 package') }
+  if (!GREXTRA)  { stop('Error: Plotting a \'ballbins\' object requires the gridExtra package') }
+
+  #Generate occupancy indicators
+  OCC           <- matrix(FALSE, nrow = obs, ncol = m)
+  rownames(OCC) <- sprintf('Sample[%s]', 1:obs)
+  colnames(OCC) <- sprintf('Bin[%s]', 1:m)
+  for (i in 1:obs) {
+    for (k in 1:m) {
+      EFF <- (!is.na(SAMPLE[i, ]))
+      OCC[i, k]   <- (sum(SAMPLE[i, EFF] == k) > 0) } }
+
+  #Set ball and bin size and colour
+  if (!is.null(ball.size))   { BALL.SIZE   <- ball.size   } else { BALL.SIZE   <- 2 }
+  if (!is.null(ball.colour)) { BALL.COLOUR <- ball.colour } else { BALL.COLOUR <- 'blue' }
+
+  #Set subtitle for plot
+  if (obs == 1) {
+    SUBTITLE <- paste0('(Showing one simulation of process with ', n, ' balls allocated to ', m,
+                       ' bins, with occupancy probability = ', round(prob, 4), ')') }
+  if (obs > 1) {
+    SUBTITLE <- paste0('(Showing ', obs, ' simulations of process with ', n, ' balls allocated to ', m,
+                       ' bins, with occupancy probability = ', round(prob, 4), ')') }
+  if (!is.null(alloc.prob)) {
+    SUBTITLE <- paste0(SUBTITLE, '\n', '(Allocation probability vector = ', round(alloc.prob, 4), ')') }
+  if (prob < 1) {
+    SUBTITLE <- paste0(SUBTITLE, '\n', '(Filled balls are those that occupy their allocated bins --- filled bins are those that are occupied)') }
+
+  #Create the ball data
+  BALLDATA <- data.frame(Sample = rep(1:obs, each = n), Ball = rep(1:n, times = obs), Bin = 0, Occ = TRUE, Type = 'Ball',
+                         Label  = rep(sprintf('Sample[%s]', 1:obs), each = n))
+  for (i in 1:(obs*n)) {
+    BALLDATA$Bin[i] <- ALLOC[BALLDATA$Sample[i], BALLDATA$Ball[i]]
+    BALLDATA$Occ[i] <- !is.na(SAMPLE[BALLDATA$Sample[i], BALLDATA$Ball[i]]) }
+  BALLDATA$Shape <- factor(rep('21', obs*n), levels = c('21', '22'))
+
+  #Create the bin data
+  BINDATA <- data.frame(Sample = rep(1:obs, each = m), Ball = n+1+(BALL.SIZE)/4, Bin = rep(1:m, times = obs), Occ = c(t(OCC)), Type = 'Bin',
+                        Label  = rep(sprintf('Sample[%s]', 1:obs), each = m))
+  BINDATA$Shape  <- factor(rep('22', obs*m), levels = c('21', '22'))
+
+  #Create the sample plots
+  PLOTDATA <- rbind(BALLDATA, BINDATA)
+  LABELS <- as.list(rownames(ALLOC))
+  names(LABELS) <- 1:obs
+  PLOT <- ggplot2::ggplot(ggplot2::aes(x = !!as.symbol("Bin"), y = !!as.symbol("Ball"), fill = !!as.symbol("Occ")), data = PLOTDATA) +
```

Comment:
Note that `aes_string` and `aes_()` are soft-deprecated, this is an unfortunately ugly workaround to resolve the global variable warning in check.